phoenix-core-client/src/main/java/org/apache/phoenix/expression/util/bson/UpdateExpressionUtils.java

```diff
@@ -0,0 +1,794 @@
+   * only on Set data structure. The document is updated by removing the given set of elements from
+   * the given set of elements.
+   * Let's say if the document field is of string set data type, and the elements are:
+   * {"yellow", "green", "red", "blue"}. The elements to be removed from the set are provided
+   * as {"blue", "yellow"} with the delete expression. The operation is expected to update the
+   * existing field by removing "blue" and "yellow" from the given set and the resultant set is
+   * expected to contain: {"green", "red"}.
+   *
+   * @param deleteExpr Delete Expression Document with key-value pairs. Key represents field in the
+   * given document, on which operation is to be performed. Value represents set of elements to be
+   * removed from the existing set.
+   * @param bsonDocument Document contents to be updated.
+   */
+  private static void executeDeleteExpression(final BsonDocument deleteExpr,
+      final BsonDocument bsonDocument) {
+    for (Map.Entry<String, BsonValue> deleteEntry : deleteExpr.entrySet()) {
+      String fieldKey = deleteEntry.getKey();
+      BsonValue newVal = deleteEntry.getValue();
+      BsonValue topLevelValue = bsonDocument.get(fieldKey);
+      if (!isBsonSet(newVal)) {
+        throw new RuntimeException("Type of new value to be removed should be sets only");
+      }
+      // If the top level field exists, perform the operation here and return.
+      if (topLevelValue != null) {
+        BsonValue value = modifyFieldValueByDelete(topLevelValue, newVal);
+        if (value == null) {
+          bsonDocument.remove(fieldKey);
+        } else {
+          bsonDocument.put(fieldKey, value);
+        }
+      } else if (!fieldKey.contains(".") && !fieldKey.contains("[")) {
+        LOGGER.info("Nothing to be removed as field with key {} does not exist", fieldKey);
+      } else {
+        // If the top level field does not exist and the field key contains "." or "[]" notations
+        // for nested document or nested array, use the self-recursive function to go through
+        // the document tree, one level at a time.
+        updateNestedField(bsonDocument, 0, fieldKey, newVal, UpdateOp.DELETE_FROM_SET);
+      }
+    }
+  }
+
+  /**
+   * Update the existing set by removing the set values that are present in
+   * {@code setValuesToDelete}. For this operation to be successful, both {@code currentValue} and
+   * {@code setValuesToDelete} must be of same set data type. For instance, both must be either
+   * set of string, set of numbers or set of binary values.
+   *
+   * @param currentValue The value that needs to be updated by performing deletion operation.
+   * @param setValuesToDelete The set values that need to be deleted from the currentValue set.
+   * @return Updated set after performing the set difference operation.
+   */
+  private static BsonValue modifyFieldValueByDelete(final BsonValue currentValue,
```

Comment:
This should be called modifyFieldValueByDeleteFromSet

Reply:
looks like i missed this, let me make this change.